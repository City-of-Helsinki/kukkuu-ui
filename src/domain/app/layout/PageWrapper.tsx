import { ReactNode } from 'react';
import classnames from 'classnames';

import PageMeta from './utilityComponents/PageMeta';
import Container from './Container';
import styles from './pageWrapper.module.scss';

type Props = {
  className?: string;
  title?: string;
  containerClassName?: string;
  description?: string;
  children: ReactNode;
};

const PageWrapper = ({
  children,
  className,
  containerClassName,
  title,
  description = 'homePage.hero.descriptionText',
}: Props) => {
  return (
    <div className={classnames(styles.pageWrapper, className)}>
      <PageMeta title={title} description={description} />
      <Container className={classnames(containerClassName)}>
        {children}
      </Container>
    </div>
  );
};

export default PageWrapper;
