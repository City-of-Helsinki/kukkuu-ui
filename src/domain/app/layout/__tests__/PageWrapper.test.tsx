import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import PageWrapper from '../PageWrapper';

it('renders snapshot correctly', () => {
  const pageWrapper = shallow(<PageWrapper />);
  expect(toJson(pageWrapper)).toMatchSnapshot();
});
