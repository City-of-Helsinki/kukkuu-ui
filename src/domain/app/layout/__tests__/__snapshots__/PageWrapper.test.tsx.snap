// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot correctly 1`] = `
<div
  className="pageWrapper"
>
  <Helmet
    defer={true}
    encodeSpecialCharacters={true}
  >
    <html
      lang="fi"
    />
    <title>
      Kulttuurin kummilapset
    </title>
    <meta
      content="Kaikki vuodesta 2020 eteenpäin syntyvät helsinkiläiset lapset kutsutaan kulttuurin kummilapsiksi. Kummilapset saavat kutsun vuosittain kahteen tapahtumaan, kunnes lapsi aloittaa koulun. Tapahtumat ovat maksuttomia."
      name="description"
    />
    <link
      href="/fi//"
      hrefLang="fi"
      rel="alternate"
    />
    <link
      href="/sv//"
      hrefLang="sv"
      rel="alternate"
    />
    <link
      href="/en//"
      hrefLang="en"
      rel="alternate"
    />
  </Helmet>
  <Container />
</div>
`;
